src/Azure/Orleans.Reminders.AzureStorage/Storage/AzureBasedReminderTable.cs

```diff
@@ -27,15 +29,37 @@ public sealed class AzureBasedReminderTable : IReminderTable
             this.loggerFactory = loggerFactory;
             this.clusterOptions = clusterOptions.Value;
             this.storageOptions = storageOptions.Value;
+            this.remTableManager = new RemindersTableManager(
+                this.clusterOptions.ServiceId,
+                this.clusterOptions.ClusterId,
+                this.storageOptions,
+                this.loggerFactory);
         }
 
-        public async Task Init()
+        public async Task StartAsync(CancellationToken cancellationToken)
         {
-            this.remTableManager = await RemindersTableManager.GetManager(
-                this.clusterOptions.ServiceId,
-                this.clusterOptions.ClusterId,
-                this.loggerFactory,
-                options: this.storageOptions);
+            while (!cancellationToken.IsCancellationRequested)
+            {
+                try
+                {
+                    await remTableManager.InitTableAsync();
+                    _initializationTask.TrySetResult();
```

Comment:
should do try set cancelled on the _initializationTask for cancelled start? that way other methods don't end up waiting indefinitely?